docs/kb/callbacks.md

```diff
@@ -50,98 +35,130 @@ q)proc:{[arg1;  ;argn ; callname]  }
 
 Define the following on the server.
 
 ```q
-q)add3:{x+y+z}
-q)marshal:{(neg .z.w) (z; (value x) . y)}
+q)\p 5000                                           / listen on port 5000
+q)add3:{x+y+z}                                      / server function
+q)marshal:{(neg .z.w) (z; (value x) . y)}           / function for client to call
 ```
 
 Here the function `marshal` expects the name of a non-unary function in the first parameter, an argument list for the wrapped function in the second argument and the name of the callback function used to pass back the result in the third argument.
 
 Now execute the following on the client.
 
 ```q
-q)(neg h) (`marshal; `add3; 1 2 3; `echo)
+q)clientFunc:{0N!x;}
+q)h:hopen `::5000
+q)(neg h) (`marshal; `add3; 1 2 3; `clientFunc)
+  ...
+q)hclose h
 ```
 
-This expression makes an async call to the remote function `marshal`, asking it to invoke the remote function `add3` with list argument `1 2 3` and to pass the result back to `echo`.
+This expression makes an async call to the remote function `marshal`, asking it to invoke the remote function `add3` with list argument `1 2 3` and to pass the result back to `clientFunc`.
 
 The result is that the list is summed on the server and then 6 is displayed on the client.
 
 
-## Example 4
+### Anonymous functions
 
-It is also possible to send a function to be executed remotely on the server using an alternative form of IPC. In this case, nothing needs to be defined on the server in advance. Here we show an example sending an anonymous function that returns its value to the client. 
+It is also possible to send a function to be executed remotely on the server using an alternative form of IPC. 
+In this case, nothing needs to be defined on the server in advance. Here we show an example sending an anonymous function that returns its value to the client. 
 
-!!! warning
+Start a kdb+ server listening on a choosen port e.g.
 
-    Give careful consideration before allowing this style IPC in a production environment as a client can bring down an unprotected server.
+```bash
+$ q -p 5000
+```
 
 Execute the following on the client.
 
 ```q
-q)(neg h) ({(neg .z.w) (z; x*y)}; 6; 7; `echo)
+q)clientFunc:{0N!x;} 
+q)h:hopen `::5000
+q)(neg h) ({(neg .z.w) (z; x*y)}; 6; 7; `clientFunc)
 ```
 
 This expression makes an async call sending: 
 
 - a function that multiplies two arguments and returns the result with a callback
 - the arguments 6 and 7
-- and the name of `echo` for the callback
+- and the name of `clientFunc` for the callback
 
 The result is that 6 and 7 are multiplied on the server and then 42 is displayed on the client.
+
+!!! warning
+
+    Give careful consideration before allowing this style IPC in a production environment as a client can bring down an unprotected server.
```

Comment:
This warning is lacking. What is the workaround? Can anyone do anything if it brings down the server? This leaves the reader with a dilemma and not much information on how to decide or resolve the problem if it happens.

Reply:
improved in subsequent commit